test/e2e/cert_manager_deployment_test.go

```diff
@@ -236,48 +236,62 @@ func TestCertRenew(t *testing.T) {
 			},
 		},
 	}
-	certmanagerv1Client, err := certmanagerclientset.NewForConfig(config)
-	require.NoError(t, err)
-	crt, err = certmanagerv1Client.CertmanagerV1().Certificates(crt.ObjectMeta.Namespace).Create(ctx, crt, metav1.CreateOptions{})
-	defer certmanagerv1Client.CertmanagerV1().Certificates(crt.ObjectMeta.Namespace).Delete(ctx, crt.ObjectMeta.Name, metav1.DeleteOptions{})
-	require.NoError(t, err)
-	err = wait.PollImmediate(PollInterval, TestTimeout, func() (bool, error) {
+	// certmanagerClient is defined in utils_test and reusing it here for waitForCertificateReadiness().
+	certmanagerClient, err = certmanagerclientset.NewForConfig(config)
+	require.NoErrorf(t, err, "failed to create cert manager client: %v", err)
+	_, err = certmanagerClient.CertmanagerV1().Certificates(crt.Namespace).Create(ctx, crt, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create certificate: %v", err)
+	defer certmanagerClient.CertmanagerV1().Certificates(crt.Namespace).Delete(ctx, crt.Name, metav1.DeleteOptions{})
+
+	err = waitForCertificateReadiness(ctx, crt.Name, crt.Namespace)
+	require.NoError(t, err, "failed while waiting for certificate to be ready: %v", err)
+
+	crt, err = certmanagerClient.CertmanagerV1().Certificates(crt.Namespace).Get(ctx, crt.Name, metav1.GetOptions{})
+	require.NoError(t, err, "failed to fetch certificate: %v", err)
+
+	// Store the NotAfter validity time of the newly created certificate.
+	expiryTime := crt.Status.NotAfter.Time.UTC()
```

Comment:
Should we use `expiryTime, err := library.GetCertExpiry(ingress_host+":443", tlsConfig.Clone()) `for consistency with testcase of validating tls certificate renewal?

Reply:
I thought to use the time updated by the operator to compare it against received certificate, so that status update is also validated. Please let me know WDYT?

Reply:
suggestion incorporated.

Reply:
@TrilokGeer ^^